### PR TITLE
add nestedVirtualization documentation to default.yaml

### PR DESCRIPTION
**Pull request by olamilekan000**
```

change documents nestedVirtualization configuration in the default.yaml file.
```

Link to issue: https://github.com/lima-vm/lima/issues/3031

### CODE REVIEW

**Comment by olamilekan000** [comment]
@AkihiroSuda  kindly review

File: templates/default.yaml [hunk 1]
```diff
@@ -534,6 +534,17 @@ guestInstallPrefix: null
 # 🟢 Builtin default: false
 plain: null
 
+# When the "nestedVirtualization" feature is enabled:
+# - Allows running a virtual machine inside the guest VM.
+# - Requires hardware and host support for nested virtualization.
+# - The guest VM must configure QEMU with the `-cpu host` parameters to enable proper functionality:
+#   qemu-system-aarch64 -accel kvm -cpu host -M virt -nographic -kernel vmlinuz-virt -initrd initramfs-virt
+# - Without this configuration, nested virtualization may fail with the error:
```
Comment (AkihiroSuda) [inline templates/default.yaml:542]:
```suggestion
# - Without specifying `-cpu host`, nested virtualization may fail with the error:
```

File: templates/default.yaml [hunk 2]
```diff
@@ -534,6 +534,17 @@ guestInstallPrefix: null
...
+#   qemu-system-aarch64: kvm_init_vcpu: kvm_arch_init_vcpu failed (0): Invalid argument
+# - Only supported on platforms where nested virtualization is explicitly enabled.
```
Comment (AkihiroSuda) [inline templates/default.yaml:544]:
```suggestion
# - Only supported on Apple M3 or later with `vmType: vz`.
```

File: templates/default.yaml [hunk 3]
```diff
@@ -534,6 +534,17 @@ guestInstallPrefix: null
 # 🟢 Builtin default: false
 plain: null
 
+# When the "nestedVirtualization" feature is enabled:
+# - Allows running a virtual machine inside the guest VM.
+# - Requires hardware and host support for nested virtualization.
+# - The guest VM must configure QEMU with the `-cpu host` parameters to enable proper functionality:
+#   qemu-system-aarch64 -accel kvm -cpu host -M virt -nographic -kernel vmlinuz-virt -initrd initramfs-virt
```
Comment (AkihiroSuda) [inline templates/default.yaml:541]: `-nographic`, `-kernel`, and `-initrd` are not necessary

File: templates/default.yaml [hunk 4]
```diff
@@ -534,6 +534,17 @@ guestInstallPrefix: null
 # 🟢 Builtin default: false
 plain: null
 
+# When the "nestedVirtualization" feature is enabled:
+# - Allows running a virtual machine inside the guest VM.
+# - Requires hardware and host support for nested virtualization.
+# - The guest VM must configure QEMU with the `-cpu host` parameters to enable proper functionality:
```
Comment (AkihiroSuda) [inline templates/default.yaml:540]:
```suggestion
# - The guest VM must configure QEMU with the `-cpu host` parameters to run a nested VM:
```

File: templates/default.yaml [hunk 5]
```diff
@@ -534,6 +534,17 @@ guestInstallPrefix: null
 # 🟢 Builtin default: false
 plain: null
 
+# When the "nestedVirtualization" feature is enabled:
+# - Allows running a virtual machine inside the guest VM.
```
Comment (AkihiroSuda) [inline templates/default.yaml:538]:
```suggestion
# - Allows running a VM inside the guest VM.
```

File: templates/default.yaml [hunk 6]
```diff
@@ -534,6 +534,17 @@ guestInstallPrefix: null
 # 🟢 Builtin default: false
 plain: null
 
+# When the "nestedVirtualization" feature is enabled:
+# - Allows running a VM inside the guest VM.
+# - Requires hardware and host support for nested virtualization.
+# - The guest VM must configure QEMU with the `-cpu host` parameters to run a nested VM:
+#   qemu-system-aarch64 -accel kvm -cpu host -M virt vmlinuz-virt initramfs-virt
```
Comment (AkihiroSuda) [inline templates/default.yaml:541]:
```suggestion
#   qemu-system-aarch64 -accel kvm -cpu host -M virt
```

File: templates/default.yaml [hunk 7]
```diff
@@ -534,6 +534,17 @@ guestInstallPrefix: null
 # 🟢 Builtin default: false
 plain: null
 
+# When the "nestedVirtualization" feature is enabled:
+# - Allows running a VM inside the guest VM.
+# - Requires hardware and host support for nested virtualization.
```
Comment (AkihiroSuda) [inline templates/default.yaml:539]: Redundant with L544 (M3)

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**